Added the KumuluzEE implementation for the schedule microservice

TFaga opened the pull request.

TFaga committed:
Added the KumuluzEE implementation for the schedule microservice

Signed-off-by: Tilen Faganel <tilen.faganel@me.com>

OndrejM commented:
I tried to build with `mvn install -DskipTests` and I got an exception when the app was started:

```
[ERROR] java.lang.NoClassDefFoundError: org/objectweb/asm/ClassVisitor
[ERROR]     a  t  org.eclipse.je tty.annotations.AnnotationConfiguration.createAnnotationParser(AnnotationConfiguration.java:506)
[ERROR]     a  t  org.eclipse.je tty.annotations.AnnotationConfiguration.scanForAnnotations(AnnotationConfiguration.java:421)
[ERROR]     a  t  org.eclipse.je tty.annotations.AnnotationConfiguration.configure(AnnotationConfiguration.java:363)
```

When I then executed the JAR from command line it was OK. So I think it's enough if the build doesn't start the app, only builds the JAR.

TFaga commented:
I see. I've put the tests to run as part of the `integration-test` phase, I'll try and move them to the `test` phase or make sure to disable them if the option is passed, so they can be skipped when installing.

OndrejM commented:
@TFaga, have you signed the Eclipse Contributor Agreement? If yes, did you use tilen.faganel@me.com with Eclipse? Otherwise, please see https://www.eclipse.org/legal/ECA.php to sign the agreement, so that we may accept your PRs.

TFaga committed:
Made sure to skip the tests for the microservice-schedule-kumuluzee module when supplying the -DskipTests property on maven build

Signed-off-by: Tilen Faganel <tilen.faganel@me.com>

TFaga commented:
I've added the tests to a profile that is disabled when -DskipTests is used. Just signed the agreement, should be OK now.

urbim committed:
Added KumuluzEE speaker microservice and updated KumuluzEE schedule microservice

zvoneg committed:
Added profile for running kumuluzee microservices.

urbim committed:
Added Microprofile Config and updated READMEs

urbim committed:
Fixed typos

TFaga commented:
We have added KumuluzEE implementations of the Schedule and the Speaker microservices. Both services have Health (`/health` endpoint), Metrics (`/metrics`) and Config included and implemented. For Metrics, there is also a Prometheus exporter accessible on `/prometheus` (which is a nice addition for the demo).

pilhuhn commented:
On 30 Sep 2017, at 4:23, Tilen Faganel wrote:

> We have added KumuluzEE implementations of the Schedule and the 
> Speaker microservices. Both services have Health (`/health` endpoint), 
> Metrics (`/metrics`) and Config included and implemented. For Metrics, 
> there is also a Prometheus exporter accessible on `/prometheus` (which 
> is a nice addition for the demo).

The MP-Metrics /metrics should respond with Prometheus
text data if nothing else is requested, so how is /prometheus
different here?

TFaga commented:
We have mapped the Prometheus to /prometheus endpoint and JSON output to /metrics endpoint, in order to better demonstrate both of the formats.
